src/ext.js

```diff
@@ -232,12 +232,46 @@ export default function ext({ translator }) {
                   expression: 'optional',
                 },
                 {
-                  component: 'color-picker',
-                  type: 'object',
-                  ref: 'style.fontColor',
-                  translation: 'properties.fontColor',
-                  dualOutput: true,
-                  show: true,
+                  type: 'items',
+                  items: {
+                    useFontColorExpression: {
+                      ref: 'style.useFontColorExpression',
+                      type: 'boolean',
+                      translation: 'Use color expression',
+                      component: 'switch',
+                      defaultValue: false,
+                      options: [
+                        {
+                          value: true,
+                          translation: 'On',
+                        },
+                        {
+                          value: false,
+                          translation: 'Off',
+                        },
+                      ],
+                    },
+                    fontColorExpression: {
+                      component: 'string',
+                      type: 'string',
+                      ref: 'style.fontColorExpression',
+                      translation: 'Expression',
+                      expression: 'optional',
```

Comment:
This makes the expression evaluate so that `hsl(x,x,x) -> RGB(y,y,y)`which we need. But this also makes it possible to set incorrect strings, without a `=`.